### PR TITLE
Added support for AAD-based authentication + authorization.

### PR DESCRIPTION
**Pull request by vcolin7**

### CODE REVIEW

File: sdk/tables/azure-data-tables/pom.xml [hunk 1]
```diff
@@ -38,6 +38,11 @@ Licensed under the MIT License.
       <artifactId>azure-core</artifactId>
       <version>1.17.0</version> <!-- {x-version-update;com.azure:azure-core;dependency} -->
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-identity</artifactId>
```
Comment (christothes) [inline sdk/tables/azure-data-tables/pom.xml:43]: Is this dependency just for the sample that includes `DefaultAzureCredential`?

File: sdk/tables/azure-data-tables/pom.xml [hunk 2]
```diff
@@ -38,6 +38,11 @@ Licensed under the MIT License.
...
+      <version>1.3.1</version> <!-- {x-version-update;com.azure:azure-identity;dependency} -->
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/pom.xml:44]: This needs to be a `<scope>test</scope>` dependency. `azure-identity` isn't allowed as a runtime dependency.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java [hunk 3]
```diff
@@ -182,4 +195,22 @@ static void validateCredentials(AzureNamedKeyCredential azureNamedKeyCredential,
...
+    /**
+     * Validates that the client is properly configured to use https.
+     *
+     * @param objectToCheck The object to check for.
+     * @param objectName The name of the object.
+     * @param endpoint The endpoint for the client.
+     */
+    public static void httpsValidation(Object objectToCheck, String objectName, String endpoint, ClientLogger logger) {
+        try {
+            if (objectToCheck != null && !new URI(endpoint).getScheme().equals(StorageConstants.HTTPS)) {
```
Comment (alzimmermsft) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java:208]: This doesn't need to be done here as the bearer token credential policy performs this validation:

```java
if ("http".equals(context.getHttpRequest().getUrl().getProtocol())) {
            return Mono.error(new RuntimeException("token credentials require a URL using the HTTPS protocol scheme"));
        }
```

File: sdk/tables/azure-data-tables/pom.xml (hunk 1 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/pom.xml:43]: Yes, but it seems I need to scope it to be included only in tests, as @alzimmermsft pointed out below.

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**